### ISSUE TITLE
Fixes for LT13 not applied, as it would result in an unparsable file

### ISSUE THREAD

**Issue by cfernhout** [sqlfluff/sqlfluff#6214]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When running `sqlfluff.fix` in Python I received the following error: 
```
Fixes for LT13 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning.
One fix for LT13 not applied, it would re-cause the same error.
```

I trimmed down my query to get to the root cause of it.

### Expected Behaviour

The leading empty line would be trimmed

E.g.
```
>>> problematic_query = """
create user some_user password '*****' default_role = some_role;
"""
>>> fixed_sql = sqlfluff.fix(problematic_query, dialect='snowflake', rules=['LT13'])
>>> print(fixed_sql)
create user some_user password '*****' default_role = some_role;

```

### Observed Behaviour

Received an error and the sql string was not fixed.

E.g.
```
>>> problematic_query = """
create user some_user password '*****' default_role = some_role;
"""
>>> fixed_sql = sqlfluff.fix(problematic_query, dialect='snowflake', rules=['LT13'])
Fixes for LT13 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning.
One fix for LT13 not applied, it would re-cause the same error.
>>> print(fixed_sql)

create user some_user password '*****' default_role = some_role;

```

### How to reproduce

```python
import sqlfluff
problematic_query = """
create user some_user password '*****' default_role = some_role;
"""
unfixed_sql = sqlfluff.fix(problematic_query, dialect='snowflake', rules=['LT13'])
```

or

```python
import sqlfluff
problematic_query = """
-- this is a comment
create user some_user password '*****' default_role = some_role;
"""
unfixed_sql = sqlfluff.fix(problematic_query, dialect='postgres', rules=['LT13'])
```

### Dialect

Tested on snowflake, redshift, postgres. Looks like it's independent of dialect.

### Version

3.1.1

### Configuration

NA

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
It appears the input sql statements are currently unable to be parsed in any of the listed dialects.

The `snowflake` version runs when using an equal operator after the `PASSWORD`
```sql
 
create user some_user password = '*****' default_role = some_role;
```

Additionally, the `postgres` dialect was also fixable using the `IN ROLE` syntax
```sql
 
create user some_user password '*****' in role some_role;
```

Please let me know if you see otherwise and we can reopen.
